python/py-pyfatfs/Portfile

```diff
@@ -17,18 +17,18 @@ long_description    ${description}
 
 homepage            https://github.com/nathanhi/pyfatfs
 
-checksums           rmd160  2b632da58d4bbea4535b23ec1f35bce3df8f5d77 \
-                    sha256  e49ea062119fdf6198c7bbbcfe08589d8919e34ac21f5f604d0ed8b5c444972d \
-                    size    50114
+checksums           rmd160  ba788728d3f2d82a0947cd4d8db7a8834fd1f3e5 \
+                    sha256  9725ccd0a4da1c09c27358abbf10f08c043ac84210af576803e087f51a2b30e0 \
+                    size    55459
 
-python.versions     38 39 310 311 312
+python.versions     39 310 311 312 313
 python.pep517       yes
 
 if {${name} ne ${subport}} {
-    patchfiles-append       patch-py312_tests.diff
     depends_build-append    port:py${python.version}-pip \
                             port:py${python.version}-setuptools_scm
     depends_lib-append      port:py${python.version}-fs
+    depends_test-append     port:py${python.version}-pytest-mock
     build.args              --skip-dependency-check
```

Comment:
this is unconventional and typically not what we want; does is not work when this line is removed?

Reply:
@reneeotten https://github.com/macports/macports-ports/pull/26329#discussion_r1819664033:
> this is unconventional and typically not what we want; does is not work when this line is removed?

No, because its [pyproject.toml pins versions](https://github.com/nathanhi/pyfatfs/blob/v1.1.0/pyproject.toml#L2) for no good reason.

```
requires = ["setuptools ~= 67.8", "setuptools_scm[toml] ~= 7.1"]
```

MacPorts’ py-setuptools is at 75.2.0, and py-setuptools_scm is at 8.1.0, so these conditions fail.

```
:info:build ERROR Missing dependencies:
:info:build     setuptools~=67.8
:info:build     setuptools_scm[toml]~=7.1
```

Recently, instead of using `--skip-dependency-check`, I’ve been `reinplace`ing these bum requirements to not be version-locked. a51eb0918203d8240318b776aba40a97756d800a, for example. Would you prefer that here?

```
    post-patch {
        reinplace -E "s|\"setuptools *\[=~<>\]\[^\"\]+\"|\"setuptools\"|" ${worksrcpath}/pyproject.toml
        reinplace -E "s|\"setuptools_scm\\\[toml\\\] *\[=~<>\]\[^\"\]+\"|\"setuptools_scm\[toml\]\"|" ${worksrcpath}/pyproject.toml
    }
```

Reply:
the preferred way is to add a patchfile that removes the version pinning - provided, of course, that the builds will still work with the newer versions that MacPorts provides.

Reply:
Is a patch file preferable to reinplace? I thought that reinplace would be better for a simple one-line change like this.

Reply:
This works fine without the pinned version. I implemented this with a patch file.